earth_enterprise/src/fusion/autoingest/sysman/AssetHandleD.h

```diff
@@ -265,26 +263,26 @@ class MutableAssetHandleD_ : public virtual Base_ {
     if (dirtyMap.size()) {
       fprintf(stderr, "========== %d dirty %s ==========\n",
               dirtyMap.size(), header.c_str());
-      for (typename DirtyMap::const_iterator d = dirtyMap.begin();
-           d != dirtyMap.end(); ++d) {
+      std::vector<std::string> dirtyKeys = dirtyMap.keys();
+      for (auto& d : dirtyKeys) {
         fprintf(stderr, "%s -> %p\n",
-                d->first.c_str(), d->second.operator->());
+                d.c_str(), dirtyMap[d].operator->());
       }
     }
   }
 
 
   static bool SaveDirtyToDotNew(khFilesTransaction &savetrans,
                                 std::vector<std::string> *saveDirty) {
-    for (typename DirtyMap::const_iterator d = dirtyMap.begin();
-         d != dirtyMap.end(); ++d) {
+    std::vector<std::string> dirtyKeys = dirtyMap.keys();
```

Comment:
The function that calls this function clears the dirtyMap after this runs, which could be a problem. We only want to clear the keys that we saved, not any that were added since the function started.